corehq/apps/userreports/tasks.py

```diff
@@ -303,119 +303,144 @@ def _queue_chunk(indicators):
 
 @quickcache(['config_id'])
 def _get_config(config_id):
-    # performance optimization for save_document. don't use elsewhere
+    # performance optimization for build_async_indicators. don't use elsewhere
     return _get_config_by_id(config_id)
 
 
 @task(queue=UCR_INDICATOR_CELERY_QUEUE, ignore_result=True, acks_late=True)
-def save_document(doc_ids):
-    lock_keys = []
-    for doc_id in doc_ids:
-        lock_keys.append(get_async_indicator_modify_lock_key(doc_id))
+def build_async_indicators(indicator_doc_ids):
+    # written to be used with _queue_indicators, indicator_doc_ids must
+    #   be a chunk of 100
 
-    indicator_config_ids = None
-    timer = TimingContext()
-    with CriticalSection(lock_keys):
-        indicators = AsyncIndicator.objects.filter(doc_id__in=doc_ids)
-        if not indicators:
-            return
+    def handle_exception(exception, config_id, docs, adapter):
+        if isinstance(exception, (ProtocolError, ReadTimeout)):
+            celery_task_logger.info("Riak error when saving config: {}".format(config_id))
```

Comment:
It might be good to change these to `datadog_counter`, something `datadog_counter('commcare.async_indicator.riak_error', 1)`, I don't think we've ever looked at these logs

Reply:
Sure, will update, I have just moved the old logging into a function.